free-programming-books-id.md

```diff
@@ -93,6 +94,8 @@
 
 * [Pemrograman Fungsional untuk Rakyat Jelata dengan Scalaz](https://leanpub.com/fpmortals-id)
 
+### Pemrograman Kompetitif
+* [Pemrograman Kompetitif Dasar](https://osn.toki.id/arsip/pemrograman-kompetitif-dasar.pdf) - OSN Toki Indonesia
```

Comment:
Could you please denote that it's a PDF at the end of the line?
Also, I think for authors, it's better to write the actual writers of the book as mentioned on the front cover, rather than the site/company the resource came from.

```suggestion
* [Pemrograman Kompetitif Dasar](https://osn.toki.id/arsip/pemrograman-kompetitif-dasar.pdf) - William Gozali & Alham Fikri Aji (PDF)
```

Reply:
Thanks for this review